pkg/kube/kube_handler.go

```diff
@@ -881,56 +861,57 @@ func ValidateOCIAppRepository(appRepo *v1alpha1.AppRepository, cli httpclient.Cl
 	return true, nil
 }
 
-func (r HelmNonOCIValidator) IsValid(cli httpclient.Client) (*ValidationResponse, error) {
+// HttpValidator is an interface for checking the validity of an AppRepo via Http requests.
+type HttpValidator interface {
+	// Validate returns a validation response.
+	Validate(cli httpclient.Client) (*ValidationResponse, error)
+}
+
+// HelmNonOCIValidator is an HttpValidator for non-OCI Helm repositories.
+type HelmNonOCIValidator struct {
+	Req *http.Request
+}
+
+func (r HelmNonOCIValidator) Validate(cli httpclient.Client) (*ValidationResponse, error) {
 
 	response := &ValidationResponse{}
 
-	res, err := cli.Do(r.req)
+	res, err := cli.Do(r.Req)
 	if err != nil {
 		// If the request fail, it's not an internal error
 		return &ValidationResponse{Code: 400, Message: err.Error()}, nil
 	}
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to parse validation response. Got: %w", err)
-	}
-	response = &ValidationResponse{Code: res.StatusCode, Message: string(body)}
+	response = &ValidationResponse{Code: res.StatusCode, Message: "OK"}
```

Comment:
This was another change that came out of the tests (checking the response). Turns out we're sending over a MB of data back with the validation request of a large repo (like bitnami) since we were putting the body in the message, the body being the index.yaml in this case. So updated to only read the body for non-200 responses from the upstream repo.